Black Clovedle/ViewController.swift

```diff
@@ -6,14 +6,710 @@
+    var genderTextArray = [String]()
+    var affiliationTextArray = [String]()
+    var magicTextArray = [String]()
+    var spiritTextArray = [String]()
+    var debutArchTextArray = [String]()
+    var characterImagesArray = [UIImage]()
+    var randomCharacter = Character.self
+    let characterNames = ["Asta", "Yuno", "Noelle Silva", "Yami Sukehiro", "Mimosa Vermillion", "Luck Voltia", "Fuegoleon Vermillion", "Nozel Silva", "Charlotte Roselei", "William Vangeance", "Julius Novachrono", "Magna Swing", "Vanessa Enoteca", "Finral Roulacase", "Gauche Adlai", "Charmy Pappitson", "Gordon Agrippa", "Grey", "Secre Swallowtail", "Klaus Lunettes", "Hamon Caseus", "Alecdora Sandler", "Letoile Becquerel", "Langris Vaude", "Rhya The Disloyal", "Licht", "Vetto The Despair", "Fana The Hatred", "Sally", "Rades Spirito", "Valtos", "Rill Boismortier", "Kirsch Vermillion", "Zora Ideale", "Dorothy Unsworth", "Leopold Vermillion", "Damnatio Kira", "Dante Zogratis", "Liebe", "Zagred", "Gadjah", "Gueldre", "Henry", "Jack The Ripper", "Kaiser Granvorka", "Lily Aquaria", "Loropechika", "Mars", "Mereoleona Vermillion", "Nacht Faust", "Nebra Silva", "Ralph Niaflem", "Sekke Bronzazza", "Sol Marron", "Solid Silva", "Undine", "Vanica", "Zenon Zogratis"]
+    var filteredData: [String] = []
+    let characterInfo = CharacterInfo()
+    let characterUpdates = Guess()
+    let greenCorrectImage = UIImage(named: "greenCorrect")
+    var correctArchNumber = 0
+    var dailyGamePlayed = UserDefaults.standard.bool(forKey: "dailyGamePlayed")
+    var currentDate = "currentDate"
+    var allTimeCorrect = UserDefaults.standard.integer(forKey: "All-Time Correct")
+    //    let characterCorrectController = CharacterCorrectViewController()
+    var currentGameDate = UserDefaults.standard.string(forKey: "currentGameDate")
+    var correctDailyCharacter = UserDefaults.standard.string(forKey: "correctDailyCharacter")
+    
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view.
+        NotificationCenter.default.addObserver(self, selector: #selector(appDidBecomeActive), name: UIApplication.didBecomeActiveNotification, object: nil)
+        submit.tintColor = .black
+        tapDismissView = UIView(frame: view.bounds)
+        tapDismissView.backgroundColor = UIColor.clear
+        tapDismissView.isHidden = true
+        tapDismissView.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(dismissKeyboardAndTable)))
+        view.addSubview(tapDismissView)
+
+        let tapGesture = UITapGestureRecognizer(target: self, action: #selector(dismissKeyboardAndTable))
+            tapGesture.cancelsTouchesInView = false
+        
+        
+        let df = DateFormatter()
+        df.dateFormat = "MM/dd"
+        currentDate = df.string(from: Date())
+        print("ths viewdidload ran")
+        //        resetGame()
+        dismissKeyboard()
+        tableView.dataSource = self
```

Comment:
This is a good place for a `configureTableView` helper function that does all the setup for the tableview. When viewDidLoad gets too long, it's hard for others to find things